src/lib/dataEntities/networks.ts

```diff
@@ -395,12 +501,46 @@ export const getEthBridgeInformation = async (
   }
 }
 
+const addNetwork = (network: Chain) => {
+  networks[network.chainID] = network
+  if (isParentChain(network)) {
+    const children = getChildrenForNetwork(network)
+    children?.forEach(child => {
+      if (child) {
+        child.partnerChainID = network.chainID
+      }
+    })
+  }
+
+  if (isChildChain(network)) {
+    const parent = networks[network.partnerChainID]
+    if (parent) {
+      parent.partnerChainIDs = [
+        ...(parent.partnerChainIDs ?? []),
+        network.chainID,
+      ]
+    } else {
+      throw new ArbSdkError(
+        `Network ${network.chainID}'s partner network, ${network.partnerChainID}, not recognized`
+      )
+    }
+  }
+
+  l1Networks = getL1Chains()
+  l2Networks = getL2Chains()
+}
+
+/**
+ * Adds custom L1 and L2 networks. These networks will be returned in `getL1Network` and `getL2Network`.
+ * @param customL1Network
+ * @param customL2Network
+ */
 export const addCustomNetwork = ({
   customL1Network,
   customL2Network,
 }: {
   customL1Network?: L1Network
-  customL2Network: L2Network
+  customL2Network: L2Network | OrbitChain
 }): void => {
```

Comment:
I think we may have to do something like this:
```
type AddCustomNetworkProps = {
  customL1Network?: L1Network
  customL2Network: L2Network
} | {
  customL1Network?: L2Network
  customL2Network: OrbitChain
}

export const addCustomNetwork = ({
  customL1Network,
  customL2Network,
}: AddCustomNetworkProps): void
```

Reply:
So either you pass in an L1 and an L2 (e.g. Sepolia and Arbitrum Sepolia) or you pass in an L2 and an Orbit chain

Again the case of misnomer, we should think of `customL1Network` as `parentChain` and `customL2Network` as `childChain`.